### PR TITLE
Add timeouts to visual regression, e2e tests, and storybook generation

### LINKED ISSUE #795: Set a timeout for Playwright jobs
## Problem

Failing Playwright tests (e2e and VR) can take up a long amount of time and consume huge amounts of action run-time. 

## Description

The API and ingestion server tests have a timeout configured so that the job is marked as failed after 15m.

https://github.com/WordPress/openverse/blob/6d949e19aab3bf47a0e799a19f231ce25fbebde6/.github/workflows/ci_cd.yml#L207

https://github.com/WordPress/openverse/blob/6d949e19aab3bf47a0e799a19f231ce25fbebde6/.github/workflows/ci_cd.yml#L156

A similar timeout should be set for the Playwright jobs (accounting for the normal time it takes for them to pass).

### PR DESCRIPTION
**Pull request by zackkrida**
## Fixes

Fixes #795 by @dhruvkb

## Description

Adds a 15 minute timeout to playwright tests, e2e tests, and storybook doc generation. This should be a sensible default timeout that leaves space for successful runs to complete.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by dhruvkb — APPROVED** [review]
LGTM. @zackkrida could you add `.github/workflows/ci_cd.yml` to the filters (https://github.com/WordPress/openverse/blob/de8a078374da546479bf534784104dfb671a222f/.github/filters.yml) for all 3 packages so that the complete CI is triggered and we can verify that 15m is enough for the tests to run completely?

File: .github/workflows/ci_cd.yml [hunk 1]
```diff
@@ -513,6 +514,7 @@ jobs:
     name: Nuxt Playwright VR tests
     if: needs.get-changes.outputs.frontend == 'true'
     runs-on: ubuntu-latest
+    timeout-minutes: 15
```
Comment (dhruvkb) [inline .github/workflows/ci_cd.yml:517]: This test takes ~12 min so 15 min might be cutting it close. Maybe a higher limit like 20 min might be more appropriate.

**Review by obulat — APPROVED** [review]
Great!
I re-checked some of the CI runs, and I guess 15 minutes should be enough.
Screenshot 2023-03-03 at 6 51 29 AM
Screenshot 2023-03-03 at 6 52 05 AM

File: .github/workflows/ci_cd.yml (hunk 1 above)
Comment (zackkrida) [reply .github/workflows/ci_cd.yml:517]: 12 is the max I've seen, with as low as 7. Could we keep this as-is and increase in future PRs which increase the duration?

**Pull request merged**